content/protocols/sse.textile

```diff
@@ -0,0 +1,230 @@
+
+In the future we plan to send an event on the connection that indicates that the token will expire imminently, which will allow a new connection to be established prior to the closure of the previous connection.
+
+h4(#specifying-lasteventid). Specifying the lastEventId
+
+Each message received will have a @lastEventId@ attribute containing the last id of any message received on the connection. When constructing a new connection, this value can be specified as a @lastEvent@ param in the URL.
+
+The following is an example of implementing message continuity with token auth:
+
+```[javascript]
+let lastEvent;
+
+const connectToAbly = () => {
+  // obtain a token
+  const token = <GET-NEW-ABLY-AUTH-TOKEN>
+
+  // establish a connection with that token
+  const lastEventParam = lastEvent ? ('&lastEvent=' + lastEvent) : '';
+  eventSource = new EventSource(`https://realtime.ably.io/sse?v=1.1&accessToken=${token}&channels=${channel}${lastEventParam}`);
+
+  // handle incoming messages
+  eventSource.onmessage = msg => {
+    lastEvent = msg.lastEventId;
+    // ... normal message processing
+  }
+
+  // handle connection errors
+  eventSource.onerror = msg => {
+    const err = JSON.parse(msg.data);
+    const isTokenErr = err.code >= 40140 && err.code < 40150;
+    if(isTokenErr) {
+      eventSource.close();
+      connectToAbly();
+    } else {
+      // ... handle other types of error -- for example, retry on 5xxxx, close on 4xxxx
+    }
+  }
+}
+
+connectToAbly();
+```
+
+An important thing to note here is that the EventSource API tries to auto-reconnect and re-subscribe to the SSE endpoint when any error occurs, even the token expiry error like in this case. This means that upon manually re-subscribing to the SSE endpoint with a new token, there will be two active subscriptions to the endpoint - one with the old token which would continue to throw an error due to expired credentials and another with the new token. Hence, it is important to close the previous @EventSource@ subscription using @eventSource.close()@ before re-subscribing with the new token as shown in the snippet above.
+
+You can take a look at a "demo app":https://sse-token-auth.glitch.me and "a complete code example for implementing message continuity in an SSE subscription when using token auth":https://glitch.com/edit/#!/sse-token-auth.
+
+h2(#channel-options). Channel options
+
+In an SSE connection, it is also possible to specify "channel options":/channels/options as a query string in the connection URL, instead of as a qualifier on an individual channel name. In this case, the given channel options apply to all channel attachments associated with that connection.
```

Comment:
I think this is facing a similar issue to my comment [here](https://github.com/ably/docs/pull/1784#discussion_r1081566456). Once you've checked that what I wrote is correct then we can update the intro here for SSE too.